lib/broccoli/glimmer-app.ts

```diff
@@ -278,19 +283,40 @@ export default class GlimmerApp extends AbstractBuild {
     return tsconfig ? { tsconfig } : DEFAULT_TS_OPTIONS;
   }
 
-  /**
-   * Creates a Broccoli tree representing the compiled Glimmer application.
-   *
-   * @param options
-   */
-  public toTree() {
-    let jsTree = this.javascriptTree();
-    let cssTree = this.cssTree();
-    let publicTree = this.publicTree();
-    let htmlTree = this.htmlTree();
-    let missingPackages = [];
+  private javascript() {
+    let { src, nodeModules } = this.trees;
+    let tsConfig = this.tsOptions();
+    let configTree = this.buildConfigTree(src);
+    let srcWithoutHBSTree = new Funnel(src, {
+      exclude: ['**/*.hbs', '**/*.ts']
+    });
+
+    // Compile the TypeScript and Handlebars files into JavaScript
+    let compiledHandlebarsTree = this.compiledHandlebarsTree(src);
+    let combinedConfigAndCompiledHandlebarsTree = new MergeTrees([configTree, compiledHandlebarsTree]);
+
+    // the output tree from typescript only includes the output from .ts -> .js transpilation
+    // and no other files from the original source tree
+    let source = this.compiledTypeScriptTree(combinedConfigAndCompiledHandlebarsTree, nodeModules, tsConfig);
```

Comment:
Can we call this `transpiledTypescriptTree`?  Calling it `source` confused me as I was thinking "raw source".

definitely not a blocking concern though